### PR TITLE
testing: Refactor plugins gtest infra to support additional test cases

### PR DESCRIPTION
**Pull request by mamar123**
## What?
Refactored plugins GTest infrastructure to support more flexible test cases and flows.

## Why?
To make it easier to add and manage test variations (e.g., sizes, random data, edge cases) and reduce duplication.

## How?
Used transferMemConfig for configuration, added support for random buffers, and simplified test setup logic.

### CODE REVIEW

**Review by ovidiusm — DISMISSED** [review]

**Review by aranadive — DISMISSED** [review]

File: test/gtest/plugins/obj_plugin.cpp [hunk 1]
```diff
@@ -50,29 +50,35 @@ class setupObjTestFixture : public setupBackendTestFixture {
...
     transfer.testTransfer(NIXL_WRITE);
-    transfer.resetLocalMem();
+    transfer.resetSrcMem();
     transfer.testTransfer(NIXL_READ);
-    transfer.checkLocalMem();
+    transfer.checkSrcMem();
 }
 
 TEST_P(setupObjTestFixture, XferMultiBufsTest) {
+    transferMemConfig mem_cfg{.numBufs_ = 3};
```
Comment (etoledano) [inline test/gtest/plugins/obj_plugin.cpp:65]: Is `3` a must? if yes, please use a constant, otherwise a random value to make it clear

File: test/gtest/plugins/transfer_handler.h [hunk 2]
```diff
@@ -26,41 +27,58 @@
...
+    std::uniform_int_distribution<> dist(min, max);
+    return dist(gen);
+}
+
+struct transferMemConfig {
+    const size_t numEntries_ = 1;
+    const size_t entrySize_ = 64;
+    const size_t numBufs_ = 1;
+    const uint8_t srcBufByte_ = getRandomInt(0, 255);
+    const uint8_t dstBufByte_ = getRandomInt(0, 255);
```
Comment (etoledano) [inline test/gtest/plugins/transfer_handler.h:43]: On second thought, not sure if `const` is the right choice when applied for everything.
This has several drawbacks e.g. impossible to assign.
If this configuration is local and we don't keep it - its fine to have the user decide.

File: test/gtest/plugins/transfer_handler.h [hunk 3]
```diff
@@ -26,41 +27,58 @@
...
-            CHECK(src_engine->supportsRemote()) << "Local engine does not support remote transfers";
-            dstDevId_ = 1;
-            verifyConnInfo();
-        } else {
-            CHECK(src_engine->supportsLocal()) << "Local engine does not support local transfers";
-            dstDevId_ = srcDevId_;
-        }
+          isRemoteXfer_(srcAgentName_ != dstAgentName_),
+          srcDevId_(0),
+          dstDevId_(isRemoteXfer_ ? 1 : 0) {
```
Comment (etoledano) [inline test/gtest/plugins/transfer_handler.h:67]: What does `0`/`1` means? should they be constants?

File: test/gtest/plugins/transfer_handler.h [hunk 4]
```diff
@@ -71,47 +89,89 @@ template<nixl_mem_t srcMemType, nixl_mem_t dstMemType> class transferHandler {
...
+            if (dstBackendEngine_->supportsProgTh()) dstBackendEngine_->progress();
+        }
+        NIXL_INFO << "\nTransfer complete";
+
+        return NIXL_SUCCESS;
+    }
+
     void
-    setLocalMem() {
+    addSrcDesc(nixlMetaDesc &meta_desc) {
```
Comment (etoledano) [inline test/gtest/plugins/transfer_handler.h:132]: can be `const`

File: test/gtest/plugins/transfer_handler.h [hunk 5]
```diff
@@ -71,47 +89,89 @@ template<nixl_mem_t srcMemType, nixl_mem_t dstMemType> class transferHandler {
...
+        srcDescs_->addDesc(meta_desc);
+    }
+
+    void
+    addDstDesc(nixlMetaDesc &meta_desc) {
```
Comment (etoledano) [inline test/gtest/plugins/transfer_handler.h:137]: can be `const`

File: test/gtest/plugins/transfer_handler.h [hunk 6]
```diff
@@ -71,47 +89,89 @@ template<nixl_mem_t srcMemType, nixl_mem_t dstMemType> class transferHandler {
...
+        nixl_status_t ret;
+        ret = srcBackendEngine_->postXfer(
+            op, *srcDescs_, *dstDescs_, dstAgentName_, xferHandle_, &xferOptArgs_);
+        return (ret == NIXL_SUCCESS || ret == NIXL_IN_PROG) ? NIXL_SUCCESS : NIXL_ERR_BACKEND;
+    }
+
+    nixl_status_t
+    waitForTransfer() {
+        nixl_status_t ret = NIXL_IN_PROG;
+        auto end_time = absl::Now() + absl::Seconds(3);
```
Comment (etoledano) [inline test/gtest/plugins/transfer_handler.h:117]: Why `3`? maybe a constant?

File: test/gtest/plugins/transfer_handler.h [hunk 7]
```diff
@@ -154,51 +214,17 @@ template<nixl_mem_t srcMemType, nixl_mem_t dstMemType> class transferHandler {
...
         }
 
-        srcDescs_ = std::make_unique<nixl_meta_dlist_t>(srcMemType);
-        dstDescs_ = std::make_unique<nixl_meta_dlist_t>(dstMemType);
-
-        int num_entries = split_buf ? NUM_ENTRIES : 1;
-        int entry_size = split_buf ? ENTRY_SIZE : BUF_SIZE;
         for (size_t i = 0; i < srcMem_.size(); i++) {
-            for (int entry_i = 0; entry_i < num_entries; entry_i++) {
+            for (size_t entry_i = 0; entry_i < memConfig_.numEntries_; entry_i++) {
```
Comment (etoledano) [inline test/gtest/plugins/transfer_handler.h:218]: `auto` is easier to maintain

File: test/gtest/plugins/transfer_handler.h [hunk 8]
```diff
@@ -71,47 +89,89 @@ template<nixl_mem_t srcMemType, nixl_mem_t dstMemType> class transferHandler {
...
+        nixl_status_t ret = NIXL_IN_PROG;
+        auto end_time = absl::Now() + absl::Seconds(3);
+
+        NIXL_INFO << "\t\tWaiting for transfer to complete...";
+        while (ret == NIXL_IN_PROG && absl::Now() < end_time) {
+            ret = srcBackendEngine_->checkXfer(xferHandle_);
+            if (ret != NIXL_SUCCESS && ret != NIXL_IN_PROG) return ret;
+
+            if (dstBackendEngine_->supportsProgTh()) dstBackendEngine_->progress();
+        }
```
Comment (etoledano) [inline test/gtest/plugins/transfer_handler.h:125]: `ret == NIXL_IN_PROG` in the condition isn't needed since you're only waiting for `NIXL_SUCCESS` or timeout
This loop can be rewritten as follows (in a more declarative way):

```
        auto progress = { if (dstBackendEngine_->supportsProgTh()) dstBackendEngine_->progress();  };
        do {
            auto ret = srcBackendEngine_->checkXfer(xferHandle_);
            if (ret != NIXL_SUCCESS && ret != NIXL_IN_PROG) return ret;

            progress();
        } while (absl::Now() < end_time);
```
